sympy/assumptions/ask.py

```diff
@@ -1325,11 +1326,16 @@ def _extract_all_facts(expr, symbol):
 
 def ask(proposition, assumptions=True, context=global_assumptions):
     """
-    Method for inferring properties about objects.
+    Function for inferring properties about objects.
 
     Explanation
     ===========
 
+    This function evaluates the *proposition* to boolean value, with
+    respect to *assumptions* and *context*. For this, multipledispatch
+    handler for the proposition predicate is first referred to. If this
+    cannot determine the result, SAT solver is used.
```

Comment:
Since this is end user documentation I think that implementation details like the fact that this uses multipledispatch don't need to be explained at the top although they could go into the remarks section below. Really it's only worth mentioning this here if we expect users to be able to implement their own handlers in which case there should be an explanation for how to do that (but not necessarily in this docstring although it could be linked from here).

Reply:
It would be clearer here to state explicitly what `ask(proposition)` and `ask(proposition, assumptions)` do. They are listed below but without explanation.